simd/2024/SIMD-06-21.md

```diff
@@ -0,0 +1,109 @@
+This works at the moment with AVX512 and will work with AVX10.2, which is officially published.
+
+AB: Hardware for this feature (AVX10.2) is not going to be available for some time.
+
+IR: As a standard I think we can count on this being available, and we can do emulation for now. 
+
+AB: Emulation is going to be very slow on current hardware.
+
+DG: We need to better define what emulation means. FP16 capabilities are important on all hardware, and what we have today in wasm is not sufficient. Andrew, I am curious what your position is, as we are working with other groups at Intel on AVX10 enablement.
+
+AB: There are a few conflicting directions here: AVX10 is one aspect, competitiveness is a different one.
+
+DG: We are looking for compute performance specifically on ML workloads. There are different directions on what different participants want from the standard, which is a difficult problem to solve evenly for everyone.
+
+AB: I feel we are playing whack-a-mole with hardware features, as we have lots of conflicting ways to extend this.
+
+DG: That is a good point, but the question is how to solve it? I.e. not adding instructions can be a disservice to our users.
+
+AB: Maybe we can do more in engines, for example how revectorization project was implemented.
+
+DG: The concern for such implementations is adoption, and we do more and more we might affect that. Spec has to be useful for the users.
+
+PP: The issue with revectorization is that pattern detection is fragile, users need have to produce kernel code that matches expected. There was a different approach floated around a few times, about defining standard fallbacks, i.e. defining extensions in terms of existing wasm instruction - user would either get hardware acceleration or a valid wasm alternative.
+
+AK: I am curious about revectorization and what handshake is needed to detect it.
+
+DG: It is kernel pattern detection, and some kernels might need to be compiled with that detection in mind.
+
+AK: That means that instead of 128-bit optimized kernels you need to produce 256-bit optimized kernels?
+
+DG: In a way, you might need to use a different vector length in the compiler.
+
+PP: XNNPACK is also detecting x86 vs Arm, if you want to extend revec to both platform you probably have to disable that check.
+
+IR: I feel people prefer to have predictable performance, and probably emulation needs to be detected somehow. Please join discussions in FP16 repo.
+
+AB: I am advocating for a more high level approach, and maybe we should investigate standard fallbacks that were mentioned. For example encryption/description functionality that can benefit from commonly available encryption instruction.
+
+DG: We are looking for better developer story, just having one API for something is very restrictive, while needing to take all possible hardware features into account is a lot of overhead. We are thinking that something 
```

Comment:
Maybe remove the second sentence, unless you have a recollection of what else has been said?